api/types.go

```diff
@@ -81,6 +81,10 @@ type Metrics struct {
 	PromptEvalDuration time.Duration `json:"prompt_eval_duration,omitempty"`
 	EvalCount          int           `json:"eval_count,omitempty"`
 	EvalDuration       time.Duration `json:"eval_duration,omitempty"`
+
+	Layers    int `json:"layers,omitempty"`
```

Comment:
I was thinking it might make sense to expose total layers and layers loaded into the GPU.

Don't we need some cmd/cli updates to get this serialized out into the `/set verbose` output?

Reply:
Expanding on this thought, the raw gpu layers number alone is probably not sufficient to really help the user understand what's going on.  What I think potentially makes sense is to calculate the total size of all layers, and then the size of the layers we load into GPU, and report a percentage of model loaded into GPU, since not all layers are the same size.  Maybe something like:
```
...
eval rate:            12.55 tokens/s
resident in gpu:      80% (3.8GB)
>>>
```

Reply:
> Don't we need some cmd/cli updates to get this serialized out into the /set verbose output?

Yes we will

>  What I think potentially makes sense is to calculate the total size of all layers, and then the size of the layers we load into GPU, and report a percentage of model loaded into GPU, since not all layers are the same size.

I'm not sure a percentage is appropriate since it might be misinterpreted as "low number bad" even though the model itself might be tiny on a GPU with a lot of VRAM

Maybe something like this?

```
vram used:  3.8 GB // actual vram offloaded to gpu
vram total: 4.2 GB // total vram we want to offload
```

Reply:
To clarify, I was thinking the percentage was not how much of the GPU's VRAM it uses, but how much of the models total size wound up in the GPU.  Small model fully loaded would say "100%" and the size in parentheses would be the ~actual VRAM used.

We might not want to expose the actual physical GPU VRAM through the API.

Reply:
Ah I misunderstood. I think we're on the same page though I'm not in favor of returning a percentage

Reply:
What about 2 metrics - predicted/calculated size (aka, size of model once loaded in [V]RAM) and size of layers and context we've actually loaded into VRAM?